Controllers/AuthController.cs

```diff
@@ -0,0 +1,57 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using PSP.Models.DTOs;
+using PSP.Models.Entities;
+using PSP.Services.Interfaces;
+
+
+namespace PSP.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class AuthController : Controller
+    {
+        private readonly ICustomersService _service;
+
+        private readonly IConfiguration _config;
+        public AuthController(ICustomersService service, IConfiguration config)
+        {
+            _service = service;
+            _config = config;
+        }
+
+        [HttpPost("login")]
+        public ActionResult Login([FromBody] CustomerLogin body)
+        {
+            var currentUser = _service.Authenticate(body);
+            if(currentUser == null) 
+            {
+                return Unauthorized("Invalid username or password");
+            }
+
+            return Ok(GenerateToken(currentUser));
+        }
+
+        private string GenerateToken(Customer customer)
```

Comment:
would be good idea to add it to separate Service, there's also `Authenticate()` in `CustomersService` service, which is related to this, but first I think it would be good to refine everything (whether we want authorization/roles, what authentication related methods we'll have, etc.) and then refactor, if we still have time before project turn in